src/otaclient/ota_core.py

```diff
@@ -245,74 +246,67 @@ def __init__(
         )
         self._downloader_mapper: dict[int, Downloader] = {}
 
-    def _calculate_delta(
-        self,
-        standby_slot_creator: StandbySlotCreatorProtocol,
-    ) -> DeltaBundle:
-        logger.info("start to calculate and prepare delta...")
-        delta_bundle = standby_slot_creator.calculate_and_prepare_delta()
+        # ------ setup OTA metadata parser ------ #
+        self._ota_metadata = OTAMetadata(
+            base_url=self.url_base,
+            session_dir=self._session_workdir,
+            ca_chains_store=ca_chains_store,
+        )
 
-        # update dynamic information
-        self._status_report_queue.put_nowait(
-            StatusReport(
-                payload=SetUpdateMetaReport(
-                    total_download_files_num=len(delta_bundle.download_list),
-                    total_download_files_size=delta_bundle.total_download_files_size,
-                    total_remove_files_num=len(delta_bundle.rm_delta),
-                ),
-                session_id=self.session_id,
-            )
+    def _download_file(self, entry: DownloadInfo) -> tuple[int, int, int]:
```

Comment:
Let's define the `@dataclass` for output.
it's more clear than `tuple[int, int, int]:`.

Reply:
Sure! As the data here is simple, I think NamedTuple is ideal.

Reply:
Updated in [d30215b](https://github.com/tier4/ota-client/pull/482/commits/d30215bf0d0762a201b5f3e64d13b357828aa1ab).